### PR TITLE
python-debug@3.12 3.12.4 (new formula)

### PR DESCRIPTION
**Pull request by XuehaiPan**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Add Python formula built with debug flags.

Reference: [Python Debug Build](https://docs.python.org/3/using/configure.html#python-debug-build): A debug build is Python built with the [--with-pydebug](https://docs.python.org/3/using/configure.html#cmdoption-with-pydebug) configure option.

### CODE REVIEW

File: Formula/p/python@3.12.rb [hunk 1]
```diff
@@ -169,7 +169,7 @@ def install
       end
 
       # Enabling LTO on Linux makes libpython3.*.a unusable for anyone whose GCC
-      # install does not match the one in CI _exactly_ (major and minßor version).
```
Comment (XuehaiPan) [inline Formula/p/python@3.12.rb:172]: There is a minor typo here.

File: Formula/p/python-debug@3.12.rb [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+      --prefix=#{prefix}
+      --enable-ipv6
+      --datarootdir=#{share}
+      --datadir=#{share}
+      --without-ensurepip
+      --enable-loadable-sqlite-extensions
+      --with-openssl=#{Formula["openssl@3"].opt_prefix}
+      --with-assertions
+      --with-pydebug
+      --with-trace-refs
```
Comment (XuehaiPan) [inline Formula/p/python-debug@3.12.rb:124]: Here are the added build flags.

File: Formula/p/python-debug@3.12.rb [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+    assert_match "PendingDeprecationWarning: message",
+                 shell_output(
+                   <<~EOS,
+                     #{python3} -Sc \
+                        'import warnings; warnings.warn("message", category=PendingDeprecationWarning)' 2>&1
+                   EOS
+                 )
+
+    # Check if sys module has member gettotalrefcount
+    system python3, "-c", "import sys; sys.gettotalrefcount()"
```
Comment (XuehaiPan) [inline Formula/p/python-debug@3.12.rb:474]: Added debug-build specific tests.

**Review by woodruffw — COMMENTED** [review]
I'll defer to other maintainers, but I'm not sure we want a dedicated formula for a debug version of Python. Adding one is likely to cause confusion for the majority of users, while only providing a small amount of convenience to a relatively small group that needs a debug build but can't produce it themselves.

As an alternative, have you considered using `pyenv` to build a local debug version? For example:

```bash
brew install pyenv
pyenv install --debug 3.12
```

**Comment by carlocab** [comment]
I agree with @woodruffw's review.

Thanks for the PR, but we're passing on this. Sorry!

**Pull request closed without merging**